tekton/publish-nightly.yaml

```diff
@@ -0,0 +1,226 @@
+
+        $(inputs.params.pathToProject)/$(outputs.resources.builtBashImage.url): busybox # image should have shell in $PATH
+        $(inputs.params.pathToProject)/$(outputs.resources.builtEntrypointImage.url): busybox # image should have shell in $PATH
+        $(inputs.params.pathToProject)/$(outputs.resources.builtGsutilImage.url): google/cloud-sdk:alpine # image should have gsutil in $PATH
+      EOF
+
+      cat /workspace/go/src/github.com/tektoncd/pipeline/.ko.yaml
+
+  - name: ensure-release-dirs-exist
+    image: busybox
+    command: ["mkdir"]
+    args:
+    - "-p"
+    - "/workspace/bucket/latest/"
+    - "/workspace/bucket/previous/"
+
+  - name: run-ko
+    # TODO(#639) we should be able to use the image built by an upstream Task here instead of hardcoding
+    image: gcr.io/tekton-nightly/ko-ci
+    env:
+    - name: KO_DOCKER_REPO
+      value: ${inputs.params.imageRegistry}
+    - name: GOPATH
+      value: /workspace/go
+    - name: CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE
+      value: /secret/release.json
+    command:
+    - /bin/sh
+    args:
+    - -ce
+    - |
+      set -e
+      set -x
+
+      # Auth with CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE
+      gcloud auth configure-docker
+
+      # ko requires this variable to be set in order to set image creation timestamps correctly https://github.com/google/go-containerregistry/pull/146
+      export SOURCE_DATE_EPOCH=`date +%s`
+
+      # Change to directory with our .ko.yaml
+      cd /workspace/go/src/github.com/tektoncd/pipeline
+
+      # Publish images and create release.yaml
+      ko resolve --preserve-import-paths -f /workspace/go/src/github.com/tektoncd/pipeline/config/ > /workspace/bucket/latest/release.yaml
+    volumeMounts:
+      - name: gcp-secret
+        mountPath: /secret
+
+  - name: generate-release-version
+    image: alpine/git
+    command:
+    - /bin/sh
+    args:
+    - -ce
+    - |
+      set -e
+      set -x
+    
+      # Can't use workingDir due to #1267
+      cd /workspace/go/src/github.com/tektoncd/pipeline
+
+      # Nightly releases use vYYYYMMDD-commit
+      COMMIT=$(git rev-parse HEAD | cut -c 1-10)
+      DATE=$(date +"%Y%m%d")
+      VERSION_TAG="$DATE-$COMMIT"
+
+      echo $VERSION_TAG > "/builder/home/version"
+
+  - name: copy-to-tagged-bucket
+    image: busybox
+    command:
+    - /bin/sh
+    args:
+    - -ce
+    - |
+      set -e
+      set -x
+
+      # Can't use workingDir due to #1267
+      cd /workspace/bucket
+
+      # TODO(https://github.com/tektoncd/triggers/issues/87) if the versionTag could be generated dynamically, we could use the same Task for nightly + official releases
+      export VERSION_TAG="$(cat /builder/home/version)"
+
+      mkdir -p /workspace/bucket/previous/$VERSION_TAG/
+      cp /workspace/bucket/latest/release.yaml /workspace/bucket/previous/$VERSION_TAG/release.yaml
+
+  - name: tag-images
```

Comment:
Kaniko (the latest version at least) supports specifying multiple destinations for a single build command, meaning that we could build and tag the images in the same step. It might be that the code would be less readable though.